reyna/src/com/b2msolutions/reyna/Dispatcher.java

```diff
@@ -244,9 +252,44 @@ private static boolean shouldGzip(Header[] headers) {
         return filteredHeaders.toArray(returnedHeaders);
     }
 
-    private static AbstractHttpEntity getCompressedEntity(String content, Context context) throws Exception {
-        byte[] data = content.getBytes();
-        return AndroidHttpClient.getCompressedEntity(data, context.getContentResolver());
```

Comment:
Can we leave the old logic as it is and check if the steam is compressed or not
boolean isZipped = new ZipInputStream(yourInputStream).getNextEntry() != null;
and remove the content-type "zip" if not?

Or just replace the function as you did but remove this check as this means we have to go through every single bytes if it is not compressed.